### DISCUSSION TITLE
Question: Distance from Point to Containing Polygon Perimeter

### DISCUSSION THREAD

**Discussion by HarelM** [NetTopologySuite/NetTopologySuite#518] [Q&A] [answered]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of NetTopologySuite
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/NetTopologySuite/NetTopologySuite/issues) and [closed](https://github.com/NetTopologySuite/NetTopologySuite/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When checking the distance between a `polygon` and a point that is within this polygon the result is 0.
While I understand the logic behind it, I was wondering if there's a way to get the actual distance to the perimeter of any geometry (basically multi polygon and polygon as the rest of the geometries would probably work as I would like).

### Steps to Reproduce
Create a polygon: (0,0), (1,0), (1,1), (0,1), (0,0)
Create a point: (0.5, 0.5)
Check the `Distance` between them.
Result: 0
I would like to find a way that will return 0.5...
I can probably get the coordinates and create a LineString and generalize this to a MultiPolygon too, but I was hoping that there was a simpler way...?

### System Configuration
Tell us about the environment where you are experiencing the bug:
- NetTopologySuite version: Latest
- Other NetTopologySuite packages and versions: N\A
- Environment (Operating system, version and so on): Any
- .NET Framework version: .net core 3.1 & 5
- Additional information:

**Comment by FObermaier** [comment]
```C#
if (polygon.Intersects(point))
    distance = polygon.Boundary.Distance(point)
```

**Comment by HarelM** [comment, marked as answer]
Thanks for the super quick response!
I was hoping there was something more elegant.
I ended up writing this method:
https://github.com/IsraelHikingMap/Site/blob/ddf9ea06eea0053ad8b9bb5fb29a7ab52264108c/IsraelHiking.API/Executors/SimplePointAdderExecutor.cs#L142
```cs
       private double DistanceWithPolygonFix(Geometry geometry, Point point)
        {
            if (geometry is Polygon polygon)
            {
                return DistanceToPolygon(polygon, point);
            }
            if (geometry is MultiPolygon multiPolygon)
            {
                return multiPolygon.OfType<Polygon>().Min(p => DistanceToPolygon(p, point));
            }
            return geometry.Distance(point);
        }

        private double DistanceToPolygon(Polygon polygon, Point point)
        {
            if (!polygon.Contains(point))
            {
                return polygon.Distance(point);
            }
            return polygon.InteriorRings.Concat(new[] { polygon.ExteriorRing }).Min(l => l.Distance(point));
        }
```
